app/components/result_summary_component.rb

```diff
@@ -0,0 +1,55 @@
+class ResultSummaryComponent < ViewComponent::Base
```

Comment:
I'm not sure the merit of using a `ViewComponent` to manipulate text maybe a `helper`?

Maybe the lack of html is a telltale sign?

Reply:
View components can build up return values via `call`, you don't need a corresponding html file - it's just a ruby class at the end of the day. Also there's a little too much logic to be put into a helper.

Reply:
Then maybe a simple `service` as it feels over engineered using a view component.

Reply:
done